### PR TITLE
HLE/APT: Implement GetAppletInfo for LLE applets.

### PR DESCRIPTION
**Pull request by Subv**
Calling this function for AppletId::Application is not yet implemented because we don't support launching applications from APT.

The HLE applet machinery is showing its age, it's getting awkward to still support it alongside LLE applets. I want to rewrite it one day to make it use an interface more similar to real applets.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3265)

### CODE REVIEW

File: src/core/hle/service/apt/apt.cpp [hunk 1]
```diff
@@ -887,6 +888,20 @@ void PreloadLibraryApplet(Service::Interface* self) {
     }
 }
 
+void FinishPreloadingLibraryApplet(Service::Interface* self) {
+    IPC::RequestParser rp(Kernel::GetCommandBuffer(), 0x17, 1, 0); // 0x00170040
+    u32 applet_id = rp.Pop<u32>();
```
Comment (B3n30) [inline src/core/hle/service/apt/apt.cpp]: in all other functions `AppletId applet_id = static_cast<AppletId>(rp.Pop<u32>());` is used

File: src/core/hle/service/apt/apt.cpp [hunk 2]
```diff
@@ -966,12 +994,27 @@ void GetAppletInfo(Service::Interface* self) {
         rb.Push(true);   // Registered
         rb.Push(true);   // Loaded
         rb.Push<u32>(0); // Applet Attributes
-    } else {
+        LOG_WARNING(Service_APT, "Using HLE applet info for applet %03X", static_cast<u32>(app_id));
+        return;
```
Comment (B3n30) [inline src/core/hle/service/apt/apt.cpp]: In this case there is nothing written to the command_buffer?

File: src/core/hle/service/apt/apt.cpp [hunk 3]
```diff
@@ -76,6 +76,7 @@ struct AppletSlotData {
     AppletId applet_id;
     AppletSlot slot;
     bool registered;
+    bool loaded;
```
Comment (B3n30) [inline src/core/hle/service/apt/apt.cpp]: Shouldn't this get initialized for every applet in Init() and also reset in Shutdown()

File: src/core/hle/service/apt/apt.cpp (hunk 2 above)
Comment (Subv) [reply src/core/hle/service/apt/apt.cpp]: It is written, the else branch was removed.

File: src/core/hle/service/apt/apt.cpp (hunk 3 above)
Comment (Subv) [reply src/core/hle/service/apt/apt.cpp]: Done

**Pull request merged**